### PR TITLE
Consider Use Of `pre-commit`

### PR DESCRIPTION
**Pull request by AFg6K7h4fhy2**
For the scope of this PR, please refer to issue #145 .

### CODE REVIEW

**Review by dylanhmorris — CHANGES_REQUESTED** [review]
Thanks @AFg6K7h4fhy2! Could you resolve the conflicts and then re-request review?

Also, some argument for using [actions/pre-commit](https://github.com/pre-commit/action) rather than the custom CFA one given that this is a public repo

**Comment by dylanhmorris** [comment]
Action still appears to be the one here [.github/actions/pre-commit/action.yaml](https://github.com/CDCgov/covid19-forecast-hub/pull/147/files#diff-90fbcc557efffbec0e9599bcca56e59c10ac129a9ceb934bde29e5ab943f747c). I was suggesting just using https://github.com/pre-commit/action

**Comment by AFg6K7h4fhy2** [comment]
Re: 

Ah, I see. From the link you sent, I was looking at [this file](https://github.com/pre-commit/action/blob/main/action.yml), rather than inside the README, which has this:

```yaml
name: pre-commit

on:
  pull_request:
  push:
    branches: [main]

jobs:
  pre-commit:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3
    - uses: actions/setup-python@v3
    - uses: pre-commit/action@v3.0.1
```

File: .github/workflows/pre-commit.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v3
```
Comment (dylanhmorris) [inline .github/workflows/pre-commit.yaml:13]:
```suggestion
    - uses: actions/checkout@v4
    - uses: actions/setup-python@v5
```

**Comment by dylanhmorris** [comment]
For the R hooks, need to setup the cache explicitly. See e.g. https://github.com/CDCgov/ww-inference-model/blob/main/.github/workflows/pre-commit.yaml

```yaml
    - uses: actions/cache@v4
      with:
        path: |
          ~/.cache/R/renv
        key: r-precommit-cache|${{ env.pythonLocation }}|${{ hashFiles('.pre-commit-config.yaml') }}
```

**Review by dylanhmorris — APPROVED** [review]
LGTM, thanks @AFg6K7h4fhy2!

**Pull request merged**